### ISSUE TITLE
Depend on webpack-manifest-plugin instead of webpack-php-manifest

### ISSUE THREAD

**Issue by colthreepv** [castiron/webpackassets-plugin#3]
Hello, this plugin might be very useful, but I think the right dependency should be https://github.com/danethurber/webpack-manifest-plugin, as it produces a simple JSON file that can be digested.

Moreover the aforementioned webpack plugin gets updated for every new webpack release (we are at 4.x now!)

**Comment by naomiyaki** [comment]
Thanks @colthreepv ! I started work on this and it's now working in the [json branch](https://github.com/castiron/webpackassets-plugin/tree/json). We used [webpack-assets-manifest](https://github.com/webdeveric/webpack-assets-manifest) instead of webpack-manifest-plugin, but the output should be pretty much the same.

We're going to try adding a few things to speed up the component before doing it as a point release, but feel free to try it now with the branch and let us know if you hit any snags!

Here's the config I used with webpack-assets-manifest so that it plays nicely with webpack-dev-server:
```
new WebpackAssetsManifest({
  publicPath(filename, manifest) {
    // Append either asset directory or dev server url
    return isProduction ?
      path.resolve("/assets/", filename) : `//localhost:8080/assets/${filename}`;
  },
  output: path.resolve(projectRoot, "www/assets/manifest.json"),
  writeToDisk: true,
  transform(assets, manifest) {
    // Add webpack-dev-server JS if running in development
    const withDev = assets;
    if (!isProduction) {
      withDev['webpack-dev-server.js'] = '//localhost:8080/webpack-dev-server.js';
    }
    return withDev;
  },
  done(manifest, stats) {
    console.log(`The asset manifest has been written to ${manifest.getOutputPath()}`);
  }
});
```

**Comment by naomiyaki** [comment]
Also, in October the interface for getting your assets out is
`{{ webpackAssets.file('filename.js') | raw }}`

**Comment by naomiyaki** [comment]
Hi @colthreepv ! We just released a v2.0 version of the plugin that uses [webpack-assets-manifest](https://github.com/webdeveric/webpack-assets-manifest) (and json) as listed above.
It works with webpack builds and dev-server, and also caches the built tags for speedy loading.
